Dockerfile.rh

```diff
@@ -40,38 +47,71 @@ LABEL name="Nexus IQ Server image" \
   io.openshift.expose-services="8071:8071" \
   io.openshift.tags="Sonatype,Nexus,IQ Server"
 
-# Mandatory parameters. Docker needs to know volume mount point and location of startup script.
-ENV SONATYPE_WORK="/sonatype-work" \
-    IQ_HOME="/opt/sonatype/nexus-iq-server/" \
-    DOCKER_TYPE="rh-docker"
-
-ARG IQ_SERVER_COOKBOOK_VERSION="release-0.4.20200928-152944.e90ad8b"
-ARG IQ_SERVER_COOKBOOK_URL="https://github.com/sonatype/chef-nexus-iq-server/releases/download/${IQ_SERVER_COOKBOOK_VERSION}/chef-nexus-iq-server.tar.gz"
-
-ADD solo.json.erb /var/chef/solo.json.erb
-
-# Chef version locked to avoid needing to accept the EULA on behalf of whomever builds the image
-RUN yum install -y --disableplugin=subscription-manager hostname procps \
-    && curl -L https://www.getchef.com/chef/install.sh | bash -s -- -v 14.12.9 \
-    && /opt/chef/embedded/bin/erb /var/chef/solo.json.erb > /var/chef/solo.json \
-    && chef-solo \
-       --recipe-url ${IQ_SERVER_COOKBOOK_URL} \
-       --json-attributes /var/chef/solo.json \
-    && rpm -qa *chef* | xargs rpm -e \
-    && rm -rf /etc/chef \
-    && rm -rf /opt/chefdk \
-    && rm -rf /var/cache/yum \
-    && rm -rf /var/chef \
-    && yum clean all
+USER root
+
+# For testing
+RUN microdnf update \
+&& microdnf install procps
+
+# Create folders
+RUN mkdir -p ${TEMP} \
+&& mkdir -m 0755 -p ${IQ_HOME} \
+&& mkdir -m 0755 -p ${SONATYPE_WORK} \
+&& mkdir -m 0755 -p ${CONFIG_HOME} \
+&& mkdir -m 0755 -p ${LOGS_HOME}
 
+# Copy config.yml and set sonatypeWork to the correct value
+COPY config.yml ${TEMP}
+RUN cat ${TEMP}/config.yml | sed -r "s/\s*sonatypeWork\s*:\s*\"?[-0-9a-zA-Z_/\\]+\"?/sonatypeWork: ${SONATYPE_WORK//\//\\/}/" > ${CONFIG_HOME}/config.yml \
+&& chmod 0644 ${CONFIG_HOME}/config.yml
+
+# Create start script
+RUN echo "/usr/bin/java ${JAVA_OPTS} -jar nexus-iq-server-${IQ_SERVER_VERSION}.jar server ${CONFIG_HOME}/config.yml 2> ${LOGS_HOME}/stderr.log" > ${IQ_HOME}/start.sh \
+&& chmod 0755 ${IQ_HOME}/start.sh
+
+# Download the server bundle, verify its checksum, and extract the server jar to the install directory
+RUN cd ${TEMP} \
+&& curl -L https://download.sonatype.com/clm/server/nexus-iq-server-${IQ_SERVER_VERSION}-bundle.tar.gz --output nexus-iq-server-${IQ_SERVER_VERSION}-bundle.tar.gz \
+&& echo "${IQ_SERVER_SHA256} nexus-iq-server-${IQ_SERVER_VERSION}-bundle.tar.gz" > nexus-iq-server-${IQ_SERVER_VERSION}-bundle.tar.gz.sha256 \
+&& sha256sum -c nexus-iq-server-${IQ_SERVER_VERSION}-bundle.tar.gz.sha256 \
+&& tar -xvf nexus-iq-server-${IQ_SERVER_VERSION}-bundle.tar.gz \
+&& mv nexus-iq-server-${IQ_SERVER_VERSION}.jar ${IQ_HOME} \
+&& cd ${IQ_HOME} \
+&& rm -rf ${TEMP}
+
+# Add group and user
+RUN groupadd nexus \
+&& adduser -d ${IQ_HOME} -c "Nexus IQ user" -g nexus -s /bin/false -r nexus
+
+# Red Hat Certified Container commands
+COPY rh-docker /
+RUN usermod -a -G root nexus \
+&& chmod -R 0755 /licenses \
```

Comment:
Q: why do we need this?

Reply:
> Q: why do we need this?

This particular line mimics what was previously being done by chef
https://github.com/sonatype/chef-nexus-iq-server/blob/601fa0417bb6a9e6fe50bfca5a29ff3453ec56d6/recipes/rh-docker.rb#L24-L29

Reply:
> > Q: why do we need this?
> 
> This particular line mimics what was previously being done by chef
> https://github.com/sonatype/chef-nexus-iq-server/blob/601fa0417bb6a9e6fe50bfca5a29ff3453ec56d6/recipes/rh-docker.rb#L24-L29

I'm assuming this directory is for storing the license? Is it correct to say with the latest version of IQ we now store the license in database?

Reply:
> I'm assuming this directory is for storing the license? Is it correct to say with the latest version of IQ we now store the license in database?

Not for the IQ license but for this license https://github.com/sonatype/chef-nexus-iq-server/blob/601fa0417bb6a9e6fe50bfca5a29ff3453ec56d6/files/rh-docker/licenses/LICENSE which is unique to the Red Hat build

Reply:
> > I'm assuming this directory is for storing the license? Is it correct to say with the latest version of IQ we now store the license in database?
> 
> Not for the IQ license but for this license https://github.com/sonatype/chef-nexus-iq-server/blob/601fa0417bb6a9e6fe50bfca5a29ff3453ec56d6/files/rh-docker/licenses/LICENSE which is unique to the Red Hat build

👍